Lib/fontTools/cffLib/specializer.py

```diff
@@ -26,26 +27,62 @@ def programToString(program):
 	return ' '.join(str(x) for x in program)
 
 
-def programToCommands(program):
+def programToCommands(program, deltasToSourceValues=None, numRegions=0):
```

Comment:
Maybe we might want to separate VF versions of `programToCommands()` and `commandsToProgram()` from the existing non-VF functions. It seems cleaner & clearer at expense of some code duplicates.

Reply:
I chose to represent the blend values as the actual master values in the 'commands' list rather than deltas, as I thought it useful to have access to this representation - there is no need to do so for technical reasons.  Doing this does require a lot more complexity in the API. I just built a version that does not convert between the delta and master values, and the changes were easy. I'm fine with going this way.

This also eliminates the need for the varStore.py::VarDataItem class in merging CFF2 charstrings. However, I'd like to keep this class anyway, as it is is useful for converting deltas to region values and back. The support for doing so with the current models.py::VariationModel class is all that is needed in this context, but there is no support for doing so when working with a built VF font and only a VarStore.

Some additional complexity is still required to convert a blend op in the 'program' list to a list with some blend values in the 'commands' list, even without the conversion of delta values. I do see that having a VF-specific version of programsToCommand() and commandsToProgram() will allow keeping the original much simpler version of these functions, but the more complex versions will still need to exist, and the developer then needs to be aware of both versions and call the right one.